modules/catalog/src/main/java/org/apache/ignite/internal/catalog/storage/UpdateLogImpl.java

```diff
@@ -165,9 +165,13 @@ private void restoreStateFromVault(OnUpdateHandler handler) {
                 break;
             }
 
+            if (appliedRevTimestamp == null) {
+                appliedRevTimestamp = metastore.appliedRevisionTimestamp();
+            }
+
             VersionedUpdate update = fromBytes(entry.value());
 
-            handler.handle(update);
+            handler.handle(update, appliedRevTimestamp);
```

Comment:
Is this logic correct? We are kind of replaying Catalog updates from the Vault (by iterating over Catalog versions), is it ok that they will all have the same timestamp?

Reply:
Good catch!